pkg/webhook/elasticquota/quota_topology_check.go

```diff
@@ -89,6 +90,37 @@ func (qt *quotaTopology) validateQuotaTopology(oldQuotaInfo, quotaInfo *QuotaInf
 	return nil
 }
 
+func (qt *quotaTopology) checkIsParentChange(oldQuotaInfo, quotaInfo *QuotaInfo, oldQuota *v1alpha1.ElasticQuota) error {
+	// means create quota, no need check
+	if oldQuotaInfo == nil || oldQuotaInfo.IsParent == quotaInfo.IsParent {
+		return nil
+	}
+
+	if len(qt.quotaHierarchyInfo[oldQuotaInfo.Name]) > 0 && !quotaInfo.IsParent {
+		return fmt.Errorf("quota has children, isParent is forbidden to modify as false, quotaName:%v", oldQuotaInfo.Name)
+	}
+
+	if quotaInfo.IsParent {
+		podList := &v1.PodList{}
+		quotaLabel := extension.LabelQuotaName
+		if oldQuota.Annotations[extension.AnnotationQuotaLabel] != "" {
+			quotaLabel = oldQuota.Annotations[extension.AnnotationQuotaLabel]
+		}
+		if err := qt.client.List(context.TODO(), podList, &client.ListOptions{
+			LabelSelector: labels.SelectorFromSet(map[string]string{
+				quotaLabel: oldQuota.Name,
```

Comment:
For this scenario, the index should be added; in addition, it is best to use utilclient.DisableDeepCopy in this scenario to reduce memory pressure.

Reply:
fixed